src/classes/PMT_Payment_TDTM.cls

```diff
@@ -313,6 +316,52 @@ public class PMT_Payment_TDTM extends TDTM_Runnable {
         }
     }
 
+    /*******************************************************************************************************
+    * @description Validate that the opportunity has the appropriate values. This will check to make sure
+    * that the opportunity amount is not set to an amount lower than the paid payments.
+    * @param newlist The list of Opportunities from trigger new.
+    * @param oldlist The list of Opportunities from trigger old.
+    * @param triggerAction  Which trigger event (BeforeInsert, AfterInsert, etc.).
+    * @return void Does not return anything
+    ********************************************************************************************************/
+    private void validateOpportunity(Map<Id, Opportunity> newMap, Map<Id, Opportunity> oldMap, TDTM_Runnable.Action triggerAction, DmlWrapper dmlWrapper) {
+        if (triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
```

Comment:
I think we should do this on "AfterUpdate".   See https://andyinthecloud.com/2015/04/19/considerations-placing-validation-code-in-an-apex-triggers/ for an explanation of why.